patches/tModLoader/Terraria.ID/BuffID.cs.patch

```diff
@@ -7,28 +7,33 @@
  
  namespace Terraria.ID
  {
-@@ -210,5 +_,24 @@
+@@ -210,5 +_,29 @@
  		public const int Oiled = 204;
  		public const int BallistaPanic = 205;
  		public const int Count = 206;
 +
-+		public static readonly IdDictionary Search = IdDictionary.Create<BuffID, short>();
++		public static readonly IdDictionary Search = IdDictionary.Create<BuffID, int>();
 +
-+		public static string GetReadableUniqueKey(int buffType) {
-+			if ( BuffID.Search.ContainsId(buffType) ) {
-+				return "Terraria " + BuffID.Search.GetName(buffType);
++		public static string GetUniqueKey(int type) {
++			if (type < 1 || type >= BuffLoader.BuffCount)
++				throw new ArgumentOutOfRangeException("Invalid type: " + type);
++			if (type < Count) {
++				return "Terraria " + BuffID.Search.GetName(type);
 +			}
-+
-+			ModBuff modBuff = BuffLoader.GetBuff(buffType);
-+			if( modBuff != null ) {
-+				return BuffID.GetReadableUniqueKey(modBuff);
-+			}
-+
```

Comment:
Why remove the ModBuff check? An out-of-range `buffType` should just mean to check if it's a ModBuff instead, right?